### DISCUSSION TITLE
How I can connect the lego spike prime hub via bluetooth in visual studio code using the pybricksdev library?

### DISCUSSION THREAD

**Discussion by tommlazzza** [pybricks/support#1014] [Q&A]
I'm trying to send some data from the hub to the terminal for some debug. But I can't figure out how I can connect the hub via Bluetooth to my PC. I tried to look in the [API documentation][1] and maybe I found something but I'm not sure.

  [1](https://docs.pybricks.com/projects/pybricksdev/en/latest/api/ble/index.html)

**Comment by dlech** [comment]
pipx run pybricksdev run ble my_program.py

**Comment by cegekaJG** [comment]
I can connect with my hub, but if my script imports any libraries, I get an error.

```
Traceback (most recent call last):
  File ".\server\app.py", line 1, in <module>
ImportError: no module named 'asyncio'
```

```py
import asyncio
from flask import Flask, jsonify, request
from controller import Controller
import broadcast
import json
from sys import argv

try:
    env = argv.pop(1)
except IndexError:
    env = 'default'
...
```

Is it possible to connect my hub within my Python script?

**Reply by laurensvalk** [reply]
> Is it possible to connect my hub within my Python script?

Does this project do what you need? https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/

**Reply by cegekaJG** [reply]
I think I can work with that. The guide mentions that it's also possible to communicate using the PyBricks API, but there are no examples. Do you know where I can find one?

**Reply by dlech** [reply]
I'm not sure what you mean. There currently aren't any communications APIs for Powered Up hubs. There are Bluetooth mailboxes for EV3 - perhaps that is what you are referring to?

**Reply by cegekaJG** [reply]
No, I am using the Mindstorms Inventor Hub. I was referring to this section: https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/#further-exploration-ble-terminal

**Reply by laurensvalk** [reply]
I would start with the example given in that tutorial.

But the example is there for the BLE terminal too. All you need to do is replace the code from [this section](https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/#handling-input-and-output-on-the-hub) with the oneliner of [this section](https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/#further-exploration-ble-terminal).

- Then, disconnect the hub from Pybricks Code.
- Connect to the hub from your own app
- Start the program on the hub by pressing the button
- All BLE data now consists of REPL "terminal" commands.
